src/runtime/lpython_builtin.py

```diff
@@ -798,6 +798,30 @@ def _lpython_str_endswith(s: str, suffix: str) -> bool:
         
     return True
 
+@overload
+def _lpython_str_partition(s:str, sep: str) -> tuple[str, str, str]:
+
+    """
+    Returns a 3-tuple splitted around seperator
+    """
+
+    if(len(s) == 0):
+        raise ValueError('empty string cannot be partitioned')
```

Comment:
Does this raise the error correctly?
If not, you can check this in the `python_ast_to_asr.cpp` itself, and do not check anything here at all.